noxfile.py

```diff
@@ -117,7 +118,11 @@ def set_dev(session):
     """Install PyBaMM in editable mode."""
     set_environment_variables(PYBAMM_ENV, session=session)
     envbindir = session.bin
-    session.install("-e", ".[all]", silent=False)
+    venv_dir = pathlib.Path('./venv').resolve()
+    session.install("virtualenv")
+    session.run("virtualenv", os.fsdecode(venv_dir), silent=True)
+    python = os.fsdecode(venv_dir.joinpath("bin/python"))
+    session.run(python, "-m", "pip", "install", "-e", ".[all]", silent=False)
```

Comment:
The purpose of the `dev` session is also to set up an instant developer environment besides getting an editable installation ready. This session should also install the `dev` dependencies. Also, we should use `external=True` to silence the warning about the Python interpreter being outside `nox`'s control.

Reply:
Another thing that can be done here is to install the `[jax,odes]` sets of extras for macOS and Linux